channels/OFF-CHAIN.md

```diff
@@ -291,23 +301,96 @@ different amounts, then that might communicate that it wants these instead.)
   blockchain to stop a cheater
 - `responder_pubkey` MUST be a valid ed25519 pubkey
 
+### `channel_reestabl`
 
-### `funding_created`
+Message code: 3
+
+It is possible to resume a channel which has been terminated either due
+to client failure or by the participants mutually agreeing to leave.
+
+```
+  name                  size (bytes)
+ ---------------------- ----
+| chain_hash           | 32 |
+ ---------------------- ----
+| channel_id           | 32 |
+ ---------------------- ----
+| length               | 16 |
+ ---------------------- ----
+| data                 | N  |
+ ---------------------- ----
+```
+
+The payload (`data`) must be the latest mutually signed offchain state, and
+the clients must verify that they have the corresponding state trees to match
+the state (otherwise, it will not be possible to close the channel later on.)
+
+In the `epoch` implementation, the state trees are cached inside the node.
+Note that if the node restarts, cached data is not likely to survive (it is
+not persisted on each update for performance reasons.) The `epoch` channel
+fsm automatically recovers the state trees and verifies that the provided
+state is in fact the last mutually signed state.
+
+#### Requirements
+
+*Responder*:
+
+- MUST abort if the `chain_id` doesn't match the current chain.
+- MUST abort if the payload does not correspond to the last known mutually
+  signed state.
+- MUST abort if it doesn't have a corresponding state tree (is able to verify
+  proof-of-inclusion) for the provided state.
+
```

Comment:
We could add something about `Responder` aborting if signature check does not match up (the `data` must be co-signed).
Not required to amend though.